src/devices/cpu/i386/i386.cpp

```diff
@@ -3944,9 +3959,76 @@ void i386_device::device_reset()
 
 	m_CPL = 0;
 
+	m_auto_clear_RF = true;
+
 	CHANGE_PC(m_eip);
 }
 
+inline void i386_device::dri_changed()
+{
+	int dr;
+	for(dr = 0; dr < 4; dr++)
+	{
+		int dr_enabled = (m_dr[7] & (1 << (dr << 1))) || (m_dr[7] & (1 << ((dr << 1) + 1))); // Check both local enable AND global enable bits for this breakpoint.
+		if(dr_enabled)
+		{
+			if(m_dr_breakpoints[dr]) m_dr_breakpoints[dr]->remove();
+			int breakpoint_type = (m_dr[7] >> ((dr << 2) + 16)) & 3;
+			int breakpoint_length = (m_dr[7] >> ((dr << 2) + 16 + 2)) & 3;
+			uint32_t phys_addr = m_dr[dr];
+			uint32_t error;
+			phys_addr = translate_address(m_CPL, TRANSLATE_READ, &phys_addr, &error);
```

Comment:
What happens if there is no mapping in the mmu for the breakpoint address?

Reply:
Most likely the CPU faults as normal :/

Reply:
Yeah, but you're not getting a correct phys_addr.  You don't want to put a breakpoint in lalaland.

Reply:
Yeah, but in that case, it would throw a host exception which would be caught by the catch block in execute_run().